examples/cc3200/xively_demo/main.c

```diff
@@ -0,0 +1,1127 @@
+    sl_WlanSet( SL_WLAN_CFG_GENERAL_PARAM_ID, 19, 1, &pValues );
+
+    // add the wlan profile
+    sl_WlanProfileAdd( ENT_NAME, strlen( ENT_NAME ), 0, &g_SecParams, 0, 1, 0 );
+
+    // set the connection policy to auto
+    pValues = 0;
+    sl_WlanPolicySet( SL_POLICY_CONNECTION, SL_CONNECTION_POLICY( 1, 1, 0, 0, 0 ),
+                      &pValues, 1 );
+
+    // wait for the WiFi connection
+    waitForWlanEvent();
+
+    // this is where the connection to Xively Broker can be established
+    ConnectToXively();
+
+    // wait for few moments
+    MAP_UtilsDelay( 80000000 );
+
+    // Disconnect from network
+    lRetVal = sl_WlanDisconnect();
+
+    //
+    // power off the network processor
+    //
+    lRetVal = sl_Stop( SL_STOP_TIMEOUT );
+
+    return SUCCESS;
+}
+
+//*****************************************************************************
+//
+//! Main function
+//!
+//! \param  None
+//!
+//! \return None
+//!
+//*****************************************************************************
+int main()
+{
+    long lRetVal = -1;
+
+    //
+    // Initialize Board configurations
+    //
+    BoardInit();
+
+    PinMuxConfig();
+
+    // configure RED and Green LED
+    GPIO_IF_LedConfigure( LED1 | LED2 | LED3 );
+
+#ifndef NOTERM
+    InitTerm();
+    ClearTerm();
+#endif
+
+    lRetVal = EntWlan();
+    if ( lRetVal < 0 )
+    {
+        ERR_PRINT( lRetVal );
+    }
+
+    LOOP_FOREVER();
+}
+
+/******************************************************************************/
+/* BEGINNING OF XIVELY CODE */
+/******************************************************************************/
+
+/* Includes used by the code below. */
+#include <time.h>
+#include <xi_bsp_rng.h>
+#include <xi_bsp_time.h>
+#include <xively.h>
+
+/******************************************************************************/
+/* BEGINNING OF XIVELY CODE GLOBAL SECTION */
+/******************************************************************************/
+
+/* Names of the MQTT topics used by this application */
+const char* const gTempratureTopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Temperature";
+const char* const gGreenLedTopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Green LED";
+const char* const gOrangeLedTopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Orange LED";
+const char* const gRedLedTopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Red LED";
+const char* const gButtonSW2TopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Button SW2";
+const char* const gButtonSW3TopicName =
+    "xi/blue/v1/" XIVELY_ACCOUNT_ID "/d/" XIVELY_DEVICE_ID "/Button SW3";
+
+/* Required by push button interrupt handlers - there is no space for user argument so we
+ * can't pass the xively's context handler as a function argument.
+ * None of the Xively's callbacks require global variables!*/
+xi_context_handle_t gXivelyContextHandle = -1;

```

Comment:
Hm... good point Arpi, but my vote would be to conform tot he standard that TI set in this file so that it's not all chaos.